### ISSUE TITLE
[expo-web-browser] Modal that opens up with openBrowserAsync(url, { createTask: false }) closes when I re-enter the app

### ISSUE THREAD

**Issue by dbrown-brightplan** [expo/expo#17776]
### Summary

A modal pops up in my app when I run `openBrowserAsync(url, { createTask: false })`. I want this modal to stay open in my app even when a user leaves the app and returns to the app.

**I noticed two different behaviors when re-entering the app:**
1. Re-entering the app through the task manager DOES NOT close the modal. This is great!
2. Re-entering the app by clicking on the app icon DOES close the modal. **This is the issue.**

I honestly don't know if case # 2 ever worked because I don't see it in the evidence for this PR that implemented the createTask param https://github.com/expo/expo/pull/12462.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### Environment

expo-env-info 1.0.3 environment info:
    System:
      OS: macOS 12.4
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 14.17.0 - ~/.nvm/versions/node/v14.17.0/bin/node
      Yarn: 1.22.17 - ~/.nvm/versions/node/v14.17.0/bin/yarn
      npm: 7.16.0 - ~/node_modules/.bin/npm
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Android Studio: 2021.1 AI-211.7628.21.2111.8309675
      Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
    npmPackages:
      babel-preset-expo: 9.1.0 => 9.1.0
      expo: ~45.0.0 => 45.0.4
      react: ^17.0.2 => 17.0.2
      react-dom: ^17.0.2 => 17.0.2
      react-native: ^0.68.2 => 0.68.2
      react-native-web: 0.17.7 => 0.17.7
      react-navigation: 4.4.4 => 4.4.4
    npmGlobalPackages:
      expo-cli: 5.4.9
    Expo Workflow: bare

### Reproducible demo

**Here is the link to the Snack:**
https://snack.expo.dev/@dbrown321/basic-webbrowser-usage

**Note:**
I literally just took the Snack from the 'Try this example on Snack' button from https://docs.expo.dev/versions/latest/sdk/webbrowser/#usage and added `{ createTask: false }` to it.

**Comment by brentvatne** [comment]
set `showInRecents: true` to keep it open

**Comment by dbrown-brightplan** [comment]
> set `showInRecents: true` to keep it open

Hi @brentvatne! With this comment, did you mean:
-  Have both `createTask: false` and `showInRecents: true` at the same time?
   - I tested this, and this is no different from what I got when I only set `createTask: false`.
   - The modal that contains the external site still closes when the user re-enters the app through the app icon.
   - It doesn't close when they re-enter the app through the task manager.

Or

- Just have `showInRecents: true`?
  - This opens the external site in a different task than the app.
  - Yes, it doesn't close the website, but now the user has to use the task manager to get to the external site.

So in both cases, the user would still have to use the task manager regardless of whether the external site opened in a new task or an in-app modal. They cannot access the external site by re-entering the app through the app icon.

**What I'm looking for is a way to open up the external site inside an in-app modal (exactly what happens with `createTask: false`), but I want this modal to stay open regardless of how the user re-enters the app (through the app icon AND the task manager).** Is there a way to do this?
